sdk/keyvault/keyvault-keys/migration-guide.md

```diff
@@ -0,0 +1,257 @@
+    - [Client constructors](#client-constructors)
+    - [Create a key](#create-a-key)
+    - [Retrieve a key](#retrieve-a-key)
+    - [List properties of keys](#list-properties-of-keys)
+    - [Delete a key](#delete-a-key)
+    - [Perform cryptographic operations](#perform-cryptographic-operations)
+* [Additional samples](#additional-samples)
+
+## Migration benefits
+
+A natural question to ask when considering whether or not to adopt a new version or library is what the benefits of doing so would be. As Azure has matured and been embraced by a more diverse group of developers, we have been focused on learning the patterns and practices to best support developer productivity and to understand the gaps that the JavaScript client libraries have.
+
+There were several areas of consistent feedback expressed across the Azure client library ecosystem. One of the most important is that the client libraries for different Azure services have not had a consistent approach to organization, naming, and API structure. Additionally, many developers have felt that the learning curve was difficult, and the APIs did not offer a good, approachable, and consistent onboarding story for those learning Azure or exploring a specific Azure service.
+
+To try and improve the development experience across Azure services, a set of uniform [design guidelines](https://azure.github.io/azure-sdk/general_introduction.html) was created for all languages to drive a consistent experience with established API patterns for all services. A set of [TypeScript specific guidelines][ts-guidelines] was also introduced to ensure that JavaScript clients have a natural and idiomatic feel with respect to the JavaScript and TypeScript ecosystems. Further details are available in the guidelines for those interested.
+
+### Cross Service SDK improvements
+
+The modern Key Vault client libraries also share some of the cross-service improvements made to the Azure development experience, such as:
+- Using the new `@azure/identity` library to share a single authentication approach between clients.
+- A unified logging and diagnostics pipeline that offers a common view of the activities across each of the client libraries.
+- (In case of JavaScript) The use of promises rather than callbacks for a simplified programming experience.
+- (In case of JavaScript) The use of async iterators in paging APIs.
+
+## Important changes
+
+### Separate packages and clients
+
+In the interest of simplifying the API `azure-keyvault` and `KeyVaultClient` were split into separate packages and clients:
+
+- [`@azure/keyvault-keys`][kvk-readme] contains `KeyClient` for working with Key Vault keys, and `CryptographyClient` for performing cryptographic operations.
+- [`@azure/keyvault-secrets`][kvs-readme] contains `SecretClient` for working with secrets.
+- [`@azure/keyvault-certificates`][kvc-readme] contains `CertificateClient` for working with certificates.
```

Comment:
Since we used "Key Vault keys" for keys, we should do the same for secrets and certificates

```suggestion
- [`@azure/keyvault-secrets`][kvs-readme] contains `SecretClient` for working with Key Vault secrets.
- [`@azure/keyvault-certificates`][kvc-readme] contains `CertificateClient` for working with Key Vault certificates.
```